codalab/server/bundle_manager.py

```diff
@@ -62,6 +62,7 @@ def run(self, sleep_time):
         logger.info('Bundle manager running!')
         while not self._is_exiting():
             try:
+                self._run_stage_bundles_in_background(sleep_time)
```

Comment:
If we want to put this in a thread, we should create a thread once at the top (not in this loop).

Reply:
I initially had the same thought. But what if something happened to this thread and then the newly submitted the jobs will never be moved from `created` to `staged` state. Then I started to put a check under the loop.